### PR TITLE
Improve Start-Sleep cmdlet to accept fractional seconds

### LINKED ISSUE #8477: Make Start-Sleep accept fractional seconds
# Summary of the new feature/enhancement

While `Start-Sleep -MilliSeconds` offers fine-grained control over the sleep duration, it would be convenient not to have two switch parameters just to be able to effectively specify _fractional_ seconds.

E.g., instead of:

```powershell
Start-Sleep -MilliSeconds 1400
```

it would be nice to be able to write:

```powershell
Start-Sleep 1.4  # wishful thinking; translate into 1,400 msecs.
```

This is in line with the `sleep` utility on macOS and Linux, which accepts fractional values (e.g., `sleep 1.4`)

Currently, fractional values are _quietly converted to `[int]` values_, which means that _rounding_ is applied, which is not only obscure behavior, but behavior that existing code is unlikely to rely upon.

As an aside: `Start-Sleep` is misnamed and should be called `Invoke-Sleep` - see #3990.

# Proposed technical implementation details (optional)

Change the data type of the `-Seconds` parameter from `[int]` to `[double]` and simply cast the internal conversion to milliseconds to `(int)`.

### PR DESCRIPTION
**Pull request by Prototyyppi**
## PR Summary
- "Seconds" parameter of Start-Sleep function is now interpreted as
  double. This removes the rounding error when the parameter with
  fractions is converted to milliseconds.
- ValidateRangeAttribute parameters converted to double to remove error
  when Seconds parameter is now of double type.

Fix https://github.com/PowerShell/PowerShell/issues/8477

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
@Prototyyppi Thanks for your first time contribution!  I've modified your description to include the issue being fixed so that it closes automatically when this PR is merged.  Since cmdlets are also public C# apis, @PowerShell/powershell-committee needs to review this before we can merge.  Because of the holidays in the US, the PS-Committee won't meet again til after the new year.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1 [hunk 1]
```diff
@@ -7,6 +7,8 @@ Describe "Start-Sleep DRT Unit Tests" -Tags "CI" {
     # the tests here are changed to be greater than 950ms.
     $minTime = 950
     $maxTime = 1200
+    $minTimeFractional = 1450
+    $maxTimeFractional = 1700
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: It is better modify existing test and not add new.

**Comment by iSazonov** [comment]
It is not a breaking change - no scripts will fail.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1 (hunk 1 above)
Comment (Prototyyppi) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: Yes this is true, I modified existing tests to test these new changes. Also fixed some indentation from the test file.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1 [hunk 2]
```diff
@@ -3,42 +3,50 @@
 Describe "Start-Sleep DRT Unit Tests" -Tags "CI" {
 
     # WaitHandle.WaitOne(milliseconds, exitContext) is not accurate.
-    # The wait time varies from 980ms to 1150ms from observation, so
-    # the tests here are changed to be greater than 950ms.
-    $minTime = 950
-    $maxTime = 1200
+    # The wait time varies from 980ms to 1150ms from observation
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: The values is not correct. Please update the comment.

Comment (Prototyyppi) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: I left the observations that were made with previous sleep time in the comment:
"The wait time varies from 980ms to 1150ms from observation when waiting for one second." 
and then how it is tested now:
 "To test fractional seconds the expected values start from 1450ms and end to 1700ms." 
How would you suggest the comment to be changed?

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: You would replace " 980ms to 1150ms" with "1450ms to 1700ms"

Comment (Prototyyppi) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: I would want to point out that to my understanding the values (980ms to 1150ms) have been derived by testing with sleep time of 1 second (as stated in the comment now). I use these observations to come up with the actual minimum and maximum values that are 500 milliseconds greater with a little safety margin. But I can change it if this is the right way to go.

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: Now we don't use old values so the comment can confuse readers.

Comment (Prototyyppi) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Start-Sleep.Tests.ps1]: This is changed now 👍

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@Prototyyppi Thanks for your contribution! Please wait PowerShell Committee approval for the public API change.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this, cmdlets are not intended to be public APIs, so no concern with taking this change

**Review by PaulHigin — APPROVED** [review]

**Pull request merged**